### PR TITLE
Adds vein type readout to multiblock miners

### PR DESCRIPTION
**Pull request by querns**
~~**NOTE**: This PR adds a soft dependency on VisualProspecting and Navigator!~~

This PR adds a readout of what vein type the multiblock miner is currently mining to both the controller GUI and the Metrics Transmitter Cover output. It can only do this if the player has Visual Prospecting installed, as it relies on querying its internal ore vein database to function. If VP isn't installed, no additional information will be added.

java_daIFaAOsWK

### CODE REVIEW

**Comment by Caedis** [comment]
VP also has a dep on GT btw

**Comment by Dream-Master** [comment]
it not builds because pre tags not allowed. When the pr is merged i can re tag it

File: dependencies.gradle [hunk 1]
```diff
@@ -47,6 +47,8 @@ dependencies {
     api('com.github.GTNewHorizons:Yamcl:0.6.0:dev')
     api('com.github.GTNewHorizons:ThaumicTinkerer:2.10.1:dev')
     api("com.github.GTNewHorizons:Mobs-Info:0.3.2-GTNH:dev")
+    api("com.github.GTNewHorizons:ForestryMC:4.9.7:dev")
```
Comment (OneEyeMaker) [inline dependencies.gradle:50]: Why do you add Forestry as `api` dependency? Especially if this mod mentioned few lines below as `implementation` dependency.

Comment (Dream-Master) [reply dependencies.gradle:50]: it was a merge conflict i guess

Comment (querns) [reply dependencies.gradle]: Yeah, this is correct. When I merged master into the PR, I got a conflict with the forestry API addition that I had to manually resolve. I didn't personally add it.

**Comment by OneEyeMaker** [comment]
Code-wise, this PR is fine. But I am concerned about the circular dependency it creates. Is there a good solution to achieve the same result without creating such dependency?

**Comment by querns** [comment]
I'll see if I can't rearchitecture it so there's a generic vein database interface in GT5U that VP registers itself with on load (and put in a thing so the generic vein database fails gracefully if nothing registered a handler with it.) Doing it this way should remove the circular dependency.

**Comment by querns** [comment]
VP dep eliminated. In its stead, I've added the aforementioned generic database that other mods can override. Once this PR is merged, I will put up a PR on Visual Prospecting to allow it to serve vein type requests.

**Review by OneEyeMaker — APPROVED** [review]

**Pull request merged**